File: truth_set/python/structure/solution-approach/standard-method/OrderProcessor.py
class OrderProcessor:
    def apply_discounts(self, order_total, discounts_applied):
        if discounts_applied > 5:
            raise Exception("Too many discounts applied.")
            
        if order_total > 100:
            order_total -= 10  
        elif order_total > 50:
            order_total -= 5
            
        return order_total

processor = OrderProcessor()

try:
    final_total = processor.apply_discounts(120, 6)
    print(f"Final Total: ${final_total}")
except Exception as error:
    print(f"Error: {str(error)}")

File: truth_set/javascript/structure/solution-approach/standard-method/OrderProcessor.js
var OrderProcessor = /** @class */ (function () {
    function OrderProcessor() {
    }
    OrderProcessor.prototype.applyDiscounts = function (orderTotal, discountsApplied) {
        if (discountsApplied > 5) {
            throw new Error("Too many discounts applied.");
        }
        if (orderTotal > 100) {
            orderTotal -= 10;
        }
        else if (orderTotal > 50) {
            orderTotal -= 5;
        }
        return orderTotal;
    };
    return OrderProcessor;
}());
var processor = new OrderProcessor();
try {
    var finalTotal = processor.applyDiscounts(120, 6);
    console.log("Final Total: $".concat(finalTotal));
}
catch (error) {
    console.error(error.message);
}